sycl/test/basic_tests/types.cpp

```diff
@@ -101,6 +101,67 @@ template <> inline void checkSizeForFloatingPoint<s::half, sizeof(int16_t)>() {
   static_assert(sizeof(s::half) == sizeof(int16_t), "");
 }
 
+template <typename vecType, int numOfElems>
+std::string vec2string(const sycl::vec<vecType, numOfElems> &vec) {
+  std::string str = "";
+  for (size_t i = 0; i < numOfElems - 1; ++i) {
+    str += std::to_string(vec[i]) + ",";
+  }
+  str = "{" + str + std::to_string(vec[numOfElems - 1]) + "}";
+  return str;
+}
+
+// the math built-in testing ensures that the vec binary ops get tested,
+// but the unary ops are only tested by the CTS tests. Here we do some
+// basic testing of the unary ops, ensuring they compile correctly.
+template <typename T> void checkVecUnaryOps(T &v) {
+
+  std::cout << vec2string(v) << std::endl;
+
+  T d = +v;
+  std::cout << vec2string(d) << std::endl;
+
+  T e = -v;
+  std::cout << vec2string(e) << std::endl;
+
+  // ~ only supported by integral types.
```

Comment:
I should update this. Not supported by floating point types is the correct formulation. std::byte is neither integral nor floating, but the unary operator~() is expected to work with it.